src/celeritas/field/CylMapField.hh

```diff
@@ -90,18 +90,9 @@ CELER_FUNCTION auto CylMapField::operator()(Real3 const& pos) const -> Real3
 
     // Convert Cartesian to cylindrical coordinates
     real_type r = hypot(pos[0], pos[1]);
-    // Ensure phi is in [0, 2\f$\pi\f$)
+    // Wrap turn from [-1/2; 1/2] into [0, 1)
     auto phi = atan2turn<real_type>(pos[1], pos[0]);
-    if (phi < zero_quantity())
-    {
-        phi.value() += 1;
-    }
-    if (CELER_UNLIKELY(phi.value() == 1))
-    {
-        // Make sure phi is in [0, 1). If phi is a negative value smaller
-        // than machine epsilon, adding 1 will result in phi equal to 1
-        phi.value() = 0;
-    }
+    phi.value() -= floor(phi.value());
```

Comment:
Won't this likely have just as many branches of the above? Have you compared the ptx output?

Reply:
PTX does seem to have [instructions for rounding](https://docs.nvidia.com/cuda/parallel-thread-execution/#rounding-modifiers)